network/test/epochtransition_test.go

```diff
@@ -31,138 +34,258 @@ import (
+}
 
-	// setup state related mocks
+// TearDownTest closes all the networks within a specified timeout
+func (suite *MutableIdentityTableSuite) TearDownTest() {
+	nets := make([]*p2p.Network, 0, len(suite.testNodes)+len(suite.removedTestNodes))
+	for _, n := range suite.testNodes {
+		nets = append(nets, n.net)
+	}
+	for _, n := range suite.removedTestNodes {
+		nets = append(nets, n.net)
+	}
+	stopNetworks(suite.T(), nets, 3*time.Second)
+}
+
+// setupStateMock setup state related mocks (all networks share the same state mock)
+func (suite *MutableIdentityTableSuite) setupStateMock() {
 	final := unittest.BlockHeaderFixture()
 	suite.state = new(mockprotocol.State)
 	suite.snapshot = new(mockprotocol.Snapshot)
 	suite.snapshot.On("Head").Return(&final, nil)
 	suite.snapshot.On("Phase").Return(flow.EpochPhaseCommitted, nil)
+	// return all the current list of ids for the state.Final.Identities call made by the network
 	suite.snapshot.On("Identities", mock.Anything).Return(
-		func(flow.IdentityFilter) flow.IdentityList { return suite.ids },
+		func(flow.IdentityFilter) flow.IdentityList {
+			return suite.testNodes.ids()
+		},
 		func(flow.IdentityFilter) error { return nil })
 	suite.state.On("Final").Return(suite.snapshot, nil)
+}
 
-	// all nodes use the same state mock
-	states := make([]*mockprotocol.State, nodeCount)
-	for i := 0; i < nodeCount; i++ {
-		states[i] = suite.state
+// addNode creates count many new nodes and appends them to the suite state variables
+func (suite *MutableIdentityTableSuite) addNode(count int) {
+	// create the ids, middlewares and networks
+	ids, mws, nets := GenerateIDsMiddlewaresNetworks(suite.T(), count, suite.logger, 100, nil, !DryRun)
+
+	// create the engines for the new nodes
+	engines := GenerateEngines(suite.T(), nets)
+
+	// create the node refreshers
+	idRefereshers := suite.generateNodeIDRefreshers(nets)
+
+	// create the test engines
+	for i := 0; i < count; i++ {
+		eng := testNode{
+			id:          ids[i],
+			mw:          mws[i],
+			net:         nets[i],
+			engine:      engines[i],
+			idRefresher: idRefereshers[i],
+		}
+		suite.testNodes = append(suite.testNodes, eng)
 	}
-
-	// create networks using the mocked state and default topology
-	sms := GenerateSubscriptionManagers(suite.T(), mws)
-	nets := GenerateNetworks(suite.T(), suite.logger, ids, mws, 100, nil, sms, !DryRun)
-	suite.nets = nets
-
-	// generate the refreshers
-	suite.idRefreshers = suite.generateNodeIDRefreshers(nets)
-
-	// generate the engines
-	suite.engines = GenerateEngines(suite.T(), nets)
 }
 
-// TearDownTest closes the networks within a specified timeout
-func (suite *MutableIdentityTableSuite) TearDownTest() {
-	stopNetworks(suite.T(), suite.nets, 3*time.Second)
+// removeNode removes a randomly chosen test node from suite.testNodes and adds it to suite.removedTestNodes
+func (suite *MutableIdentityTableSuite) removeNode() testNode {
+	// choose a random node to remove
+	i := rand.Intn(len(suite.testNodes) - 1)
```

Comment:
```suggestion
	i := rand.Intn(len(suite.testNodes))
```

`Intn(n)` is non-inclusive of `n`